pandas/core/indexes/base.py

```diff
@@ -1340,17 +1345,40 @@ def set_names(self, names, level=None, inplace=False):
     def rename(self, name, inplace=False):
         """
         Set new names on index. Defaults to returning new index.
+        Length of names must match number of levels in MultiIndex.
 
         Parameters
         ----------
-        name : str or list
-            name to set
+        name : str or sequence
```

Comment:
is there a reason you changed 'list' to 'sequence' ? 
I know it technically doest accept any list, but in index names there *can* be a difference between a list (multiple names) and a tuple (single name), so I would rather keep it as list. 
But it is true that a name is not limited to a string, so maybe we should use the more general "label" (and then "label, or list of labels")

Reply:
The reason why I used ‘sequence’ is that rename function takes not only list but also tuple in multiple names. 
I am not sure it is intended, but it is.

Reply:
Yes, it is intended. But as I tried to explain, a tuple can also be a single name (so you can eg give a list of tuples). Therefore, I don't think the change is an improvement in clarity for the user. 
I would strictly use 'list' for referring to the ability to give multiple names.

Reply:
but it works like this.
```python
>>> idx = pd.MultiIndex(levels=[[1, 2], ['one', 'two']],\
...                     labels=[[0, 0, 1, 1], [0, 1, 0, 1]],\
...                     names=['bar', None])
>>> idx
MultiIndex(levels=[[1, 2], ['one', 'two']],
           labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
           names=['bar', None])
>>> idx.rename(('foo', 'bar'))
MultiIndex(levels=[[1, 2], ['one', 'two']],
           labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
           names=['foo', 'bar'])
```

Reply:
Yes, it know it works, but we should not encourage that. 
For a single Index, there is a difference:

```
In [7]: idx = pd.Index([1, 2, 3, 4], name='foo')

In [8]: idx.rename(('bar', 'test'))
Out[8]: Int64Index([1, 2, 3, 4], dtype='int64', name=('bar', 'test'))

In [9]: idx.rename(['bar', 'test'])
...
TypeError: Int64Index.name must be a hashable type
```

So I would clearly state in our docstring one should use a list for multiple names (and not a sequence in general), to not confuse between those.

Reply:
oh, I got it. so `list` means not only data type. thank you.